spec/grape/validations/instance_behaivour_spec.rb

```diff
@@ -0,0 +1,43 @@
+require 'spec_helper'
+
+describe 'Validator with instance variables' do
+  before do
+    validator_type = Class.new(Grape::Validations::Base) do
+      def name
+        validator_const_name
+      end
+
+      def validate_param!(_attr_name, _params)
+        if @instance_variable
+          raise Grape::Exceptions::Validation, params: ['params'],
+                                               message: 'This should never happen'
+        end
+        @instance_variable = true
+      end
+    end
+    Grape::Validations.register_validator('instance_validator', validator_type)
+  end
+
+  after do
+    Grape::Validations.deregister_validator('instance_validator')
+  end
+
+  let(:app) do
+    Class.new(Grape::API) do
+      params do
+        optional :param_to_validate, instance_validator: true
+      end
+      get do
+        'noop'
+      end
+    end
+  end
+
+  it 'passes validation every time' do
+    get '/', param_to_validate: 'value'
```

Comment:
This was a good one. I added some more expectations to the test at the same time.